reference-implementation/lib/abstract-ops/readable-streams.js

```diff
@@ -422,6 +451,234 @@ function ReadableStreamTee(stream, cloneForBranch2) {
   return [branch1, branch2];
 }
 
+function ReadableByteStreamTee(stream) {
+  assert(ReadableStream.isImpl(stream));
+  assert(ReadableByteStreamController.isImpl(stream._controller));
+
+  let reader = AcquireReadableStreamDefaultReader(stream);
+  let reading = false;
+  let canceled1 = false;
+  let canceled2 = false;
+  let reason1;
+  let reason2;
+  let branch1;
+  let branch2;
+
+  const cancelPromise = newPromise();
+
+  function forwardReaderError(thisReader) {
+    uponRejection(thisReader._closedPromise, r => {
+      if (reader !== thisReader) {
+        return;
+      }
+      ReadableByteStreamControllerError(branch1._controller, r);
+      ReadableByteStreamControllerError(branch2._controller, r);
+      if (canceled1 === false || canceled2 === false) {
+        resolvePromise(cancelPromise, undefined);
+      }
+    });
+  }
+
+  function pullWithDefaultReader() {
+    if (ReadableStreamBYOBReader.isImpl(reader)) {
+      assert(reader._readIntoRequests.length === 0);
+      ReadableStreamReaderGenericRelease(reader);
+
+      reader = AcquireReadableStreamDefaultReader(stream);
+      forwardReaderError(reader);
+    }
+
+    const readRequest = {
+      chunkSteps: chunk => {
+        // This needs to be delayed a microtask because it takes at least a microtask to detect errors (using
+        // reader._closedPromise below), and we want errors in stream to error both branches immediately. We cannot let
+        // successful synchronously-available reads get ahead of asynchronously-available errors.
+        queueMicrotask(() => {
+          reading = false;
+
+          const chunk1 = chunk;
+          let chunk2 = chunk;
+          if (canceled1 === false && canceled2 === false) {
+            chunk2 = new chunk.constructor(chunk);
```

Comment:
Ugh, right, I'm trying to be too clever. This constructor works for typed arrays, but not for `DataView`s. 🤦‍♂️

I guess I'll have to write a helper function. 😛